corehq/apps/change_feed/producer/postgres.py

```diff
@@ -0,0 +1,61 @@
+import select
+import json
+import psycopg2.extensions
+from kafka import KeyedProducer
+from pillowtop.feed.interface import ChangeMeta
+
+from ..data_sources import get_document_store, POSTGRES
+from ..utils import send_to_kafka
+
+CHANNELS = {
+    XFormChannel.name: XFormChannel
+}
+
+
+class PostgresProducer(object):
+    """
+    Producer that pushes changes to Kafka
+    """
+
+    def __init__(self, kafka, data_source_type, data_source_name):
+        self._kafka = kafka
+        self._producer = KeyedProducer(self._kafka)
+        self._conn = get_document_store(POSTGRES)
+        self._conn.set_isolation_level(psycopg2.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+        self._data_source_type = data_source_type
+        self._channel = CHANNELS[data_source_name]()
+
+    def listen(self):
+        # Each instance of PostgresProducer will take up one connection to the master database. Keep these to
+        # a minimum
```

Comment:
when i think about this some more, i think we have to have a connection to each shard that is receiving data. i don't know if it'll work if we just listen to the proxy database. i think if we start to accumulate a lot of these, it's definitely a performance hit. not sure how to get around that though.

are you suggesting that the shards could directly publish to kafka? and there wouldn't have to be any app layer producer logic?